### PR TITLE
feat(installation): optional url to install a wsl rootfs from

### PR DESCRIPTION
**Pull request by didrocks**
Optional url to a WSL rootfs image we will download and import from. It will be registered as "id".
If this entry is not set, Ubuntu Pro for WSL will download a package from the Microsoft store matching "id".

UDENG-2554

### CODE REVIEW

File: hostagent.proto [hunk 1]
```diff
@@ -64,6 +64,16 @@ message Command {
     message Install {
         string id = 1;
         optional string cloudinit = 2;  // cloudinit is the yaml configuration to optionally pass to an instance.
+        /*
+          Optional url to a WSL rootfs image we will download and import from. It will be registered as "id".
+          If this entry is not set, Ubuntu Pro for WSL will download a package from the Microsoft store matching "id".
```
Comment (EduardGomezEscandell) [inline hostagent.proto]: Not a package matching the ID (e.g. `Ubuntu-Preview` is `Ubuntu (Preview)` on the store), but rather matching the `Name` as defined in https://github.com/microsoft/WSL/blob/master/distributions/DistributionInfo.json

Comment (didrocks) [reply hostagent.proto]: pushed an updated comment, tell me what you think :)

**Review by EduardGomezEscandell — APPROVED** [review]
Looks great from the WSL side :)

Let's see if the Landscape team has any comments

**Review by EduardGomezEscandell — APPROVED** [review]

File: hostagent.proto [hunk 2]
```diff
@@ -64,6 +64,17 @@ message Command {
...
+        /*
+          Optional url to a WSL rootfs image we will download and import from. It will be registered as "id".
+          If this entry is not set, Ubuntu Pro for WSL will download a package from the Microsoft store matching WSL
+          distribution name as available in wsl --install.
+        */
+        optional Rootfs rootfs = 3;
+
+        message Rootfs {
+            string url = 1;
+            optional string sha256sum = 2;
```
Comment (cnihelton) [inline hostagent.proto]: We should mention that this must be base64 encoded.

Comment (didrocks) [reply hostagent.proto]: should it? there is no binary data to pass no?

Comment (EduardGomezEscandell) [reply hostagent.proto]: The checksum is binary

Comment (jibel) [reply hostagent.proto]: The checksum is an hexadecimal string.

Comment (didrocks) [reply hostagent.proto]: I am probably missing something, but the checksum is generally hexadecimal as in https://pkg.go.dev/crypto/sha256#Sum256 example.

Comment (cnihelton) [reply hostagent.proto]: The slice of bytes confused me :) Printing with "%s" shows strange output, but "%x" clarifies that it's just a hex-string.

Thanks for the clarification.

**Review by wck0 — APPROVED** [review]
LGTM+1

**Pull request merged**